# Pandas
Code by: @jkmloom

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('Salaries.csv')

In [4]:
print(df.head(2))

   Id    EmployeeName                                        JobTitle  \
0   1  NATHANIEL FORD  GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY   
1   2    GARY JIMENEZ                 CAPTAIN III (POLICE DEPARTMENT)   

     BasePay  OvertimePay   OtherPay  Benefits   TotalPay  TotalPayBenefits  \
0  167411.18         0.00  400184.25       NaN  567595.43         567595.43   
1  155966.02    245131.88  137811.38       NaN  538909.28         538909.28   

   Year  Notes         Agency  Status  
0  2011    NaN  San Francisco     NaN  
1  2011    NaN  San Francisco     NaN  


In [5]:
# dataset info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 148654 entries, 0 to 148653
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Id                148654 non-null  int64  
 1   EmployeeName      148654 non-null  object 
 2   JobTitle          148654 non-null  object 
 3   BasePay           148045 non-null  float64
 4   OvertimePay       148650 non-null  float64
 5   OtherPay          148650 non-null  float64
 6   Benefits          112491 non-null  float64
 7   TotalPay          148654 non-null  float64
 8   TotalPayBenefits  148654 non-null  float64
 9   Year              148654 non-null  int64  
 10  Notes             0 non-null       float64
 11  Agency            148654 non-null  object 
 12  Status            0 non-null       float64
dtypes: float64(8), int64(2), object(3)
memory usage: 14.7+ MB


In [6]:
# average BasePay
print(df['BasePay'].mean())

66325.4488404877


In [7]:
# higest amount of OvertimePay in the dataset
print(df['OvertimePay'].max())

245131.88


In [8]:
# search job title of JOSEPH DRISCOLL
df[df['EmployeeName'] == 'JOSEPH DRISCOLL']['JobTitle']

24    CAPTAIN, FIRE SUPPRESSION
Name: JobTitle, dtype: object

In [9]:
# How much JOSEPH DRISCOLL make (including benefits)
df[df['EmployeeName'] == 'JOSEPH DRISCOLL']['TotalPayBenefits']

24    270324.91
Name: TotalPayBenefits, dtype: float64

In [12]:
# name of the higest paid person (including benefits): method 1
print(df[df['TotalPayBenefits'] == df['TotalPayBenefits'].max()]['EmployeeName'])

0    NATHANIEL FORD
Name: EmployeeName, dtype: object


In [13]:
# name of the higestpaid person (including bebfits): method 2
df.loc[df['TotalPayBenefits'].idxmax()]['EmployeeName']

'NATHANIEL FORD'

In [15]:
# name of lowest paid person (including benefits)
df.loc[df['TotalPayBenefits'].idxmin()]

Id                                      148654
EmployeeName                         Joe Lopez
JobTitle            Counselor, Log Cabin Ranch
BasePay                                    0.0
OvertimePay                                0.0
OtherPay                               -618.13
Benefits                                   0.0
TotalPay                               -618.13
TotalPayBenefits                       -618.13
Year                                      2014
Notes                                      NaN
Agency                           San Francisco
Status                                     NaN
Name: 148653, dtype: object

**Joe Lopez**'s total salary is in negative values (-618)

In [18]:
df.groupby('Year')['BasePay'].mean()

Year
2011    63595.956517
2012    65436.406857
2013    69630.030216
2014    66564.421924
Name: BasePay, dtype: float64

In [19]:
# How many unique job titles are there
df['JobTitle'].nunique()

2159

In [20]:
# top 5 most common jobs
df['JobTitle'].value_counts().head(5)

JobTitle
Transit Operator                7036
Special Nurse                   4389
Registered Nurse                3736
Public Svc Aide-Public Works    2518
Police Officer 3                2421
Name: count, dtype: int64

In [23]:
# How many job titles were represented by only one person in 2013?
sum(df[df['Year'] == 2013]['JobTitle'].value_counts() == 1)

202

In [25]:
# How many people have the word Chief in their job title?
def chief_strnig(title):
    if 'chief' in title.lower().split():
        return True
    else:
        return False

In [30]:
sum(df['JobTitle'].apply(lambda x: chief_strnig(x)))

477

In [32]:
# Is there a relation between the length of the job title string and salary?
df['title_len'] = df['JobTitle'].apply(len)

In [36]:
df[['TotalPayBenefits', 'title_len']].corr()

TotalPayBenefits  title_len
TotalPayBenefits          1.000000  -0.036878
title_len                -0.036878   1.000000